test/msw-api/cli/init.test.ts

```diff
@@ -44,18 +45,39 @@ test('copies the worker script into an existing directory without errors', async
   await cleanup()
 })
 
-test('throws an exception and does not copy the worker script given a non-existing directory', async () => {
+test('creates the directory if it does not exist', async () => {
+  const { prepare, getPath, cleanup } = createTeardown(
+    'tmp/cli/init/non-existing',
+  )
+  await prepare()
+
+  const { stderr } = await promisifyChildProcess(
+    exec(`node cli/index.js init ${getPath('non-existing-public')} --no-save`),
+  )
+
+  // Does not produce any errors.
+  expect(stderr).toBe('')
+  expect(
+    fs.existsSync(getPath('non-existing-public/mockServiceWorker.js')),
+  ).toBe(true)
+
+  await cleanup()
+})
+
+test('throws an exception and does not copy the worker script if creating the non-existing directory fails', async () => {
   const { prepare, getPath, cleanup } = createTeardown(
     'tmp/cli/init/non-existing',
   )
   await prepare()
 
+  fs.chmodSync('tmp/cli/init', '0444')
```

Comment:
This worked to make the mkdirSync fail locally, but it doesn't seem to work on the CI. 

Is there another/better way to test this?

Reply:
We can step one level higher in the execution stack and mock `fs.promises.mkdir` to always throw in this particular test. That will reject the promise, which would happen in case creating a directory fails (no matter the reason).

```js
import * as fs from 'fs'

afterEach(() => {
  jest.restoreAllMocks()
})

it('throws an exception when creating a worker directory failed', async () => {
  jest.spyOn(fs.promises, 'mkdir').mockRejectedValue(new Error('Original error message')

  // the rest of the test...
})
```

We should also add the original promise rejection error to our own error message. This provides more useful info to developers to debug possible issues.

Reply:
> We can step one level higher in the execution stack and mock fs.promises.mkdir to always throw in this particular test.

That was my first idea as well, but since we are starting the CLI in a child-process the fs that you mock is not the same one as the one running in that child-process. 

> We should also add the original promise rejection error to our own error message. This provides more useful info to developers to debug possible issues.

Yeah, good idea, I will add this 👍